src/EditorFeatures/VisualBasicTest/TypeInferrer/TypeInferrerTests.vb

```diff
@@ -736,5 +736,31 @@ End Module"
 End Class"
             Await TestAsync(text, "System.Object", testNode:=False, testPosition:=True)
         End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.TypeInferenceService)>
+        Public Async Function TestWhereCall() As Task
+            Dim text =
+"imports System.Collections.Generic
+class C
+    sub Foo()
+        [|ints|].Where(function(i) i > 10)
+    end sub
+end class"
+            Await TestAsync(text, "Global.System.Collections.Generic.IEnumerable(Of System.Int32)", testPosition:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.TypeInferenceService)>
+        Public Async Function TestWhereCall2() As Task
+            Dim text =
+"imports System.Collections.Generic
+class C
+    sub Foo()
+        [|ints|].Where(function(i)
+                return i > 10
+            end function)
+    end sub
+end class"
+            Await TestAsync(text, "Global.System.Collections.Generic.IEnumerable(Of System.Int32)", testPosition:=False)
```

Comment:
As above, can you add a test for something like `Dim x as Boolean = Boolean.$$`?